### PR TITLE
Add siddhi-execution-list extesnion

### PR DESCRIPTION
**Pull request by niveathika**
## Purpose
Process ArrayList objects in Siddhi.

## Goals
Add extensions support to process ArrayList objects in Siddhi

## Approach
Following extensions are introduced,
1. create (Function) - Create a list
2. add (Function) - add value to the end of the list/ by index
3. addAll (Function) - Add elements from one list to another
4. get (Function) - Get an element by index
5. isList (Function) - Checks whether the object is a list
6. isEmpty(Function)- Checks if the list is empty or not
7. size (Function) - Get the size of the list
8. contains(Function) - Checks if the list contains the element
9. containsAll(Function) - Checks if all elements from one list are present in another
10. remove (Function) - remove the given element if present
11. removeAll(Function) - Remove all elements from one list in another
12. removeByIndex() - Remove the given index
13. retainAll() - Retain all the elements in one list 
14. setValue() to support list replace function
15. indexOf(Function) - Returns the index of an element 
16. lastIndexOf(Function) - Returns the last index of an element in case of duplicates
17. tokenize (Stream Processor) - to tokenize the list and return each value as new attributes in events, 
18. collect  (Aggregate Function) - to collect multiple values and construct a list, Also, support distinct values
19. clear - Clear a list

## User stories
https://groups.google.com/forum/?hl=en#!topic/siddhi-dev/C2aF70NjFEc

## Release note
Extension that provides the capability to send an array object inside Siddhi stream definitions and use it within queries 

## Documentation
Attached Herewith

## Automation tests
 - Unit tests - Attached herewith
 - Integration tests - N/A

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: checkstyle-suppressions.xml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mohanvive) [inline checkstyle-suppressions.xml:1]: Do we need this checkstyle suppression  ?

File: component/src/main/java/io/siddhi/extension/execution/list/CloneFunctionExtension.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c)  2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (mohanvive) [inline component/src/main/java/io/siddhi/extension/execution/list/CloneFunctionExtension.java]: Change the license year.

**Review by mohanvive — CHANGES_REQUESTED** [review]

File: checkstyle-suppressions.xml (hunk 1 above)
Comment (niveathika) [reply checkstyle-suppressions.xml:1]: No, it if from the archetype, removed in 200938d

File: component/src/main/java/io/siddhi/extension/execution/list/CloneFunctionExtension.java (hunk 2 above)
Comment (niveathika) [reply component/src/main/java/io/siddhi/extension/execution/list/CloneFunctionExtension.java]: Fixed in 200938d

**Review by mohanvive — APPROVED** [review]

**Pull request merged**